File: 프로그래머스/unrated/181897. 리스트 자르기/리스트 자르기.java
class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        int start = 1 < n ? slicer[0] : 0;
        int end = n == 2 ? num_list.length - 1 : slicer[1];
        int diff = n == 4 ? slicer[2] : 1;
        int len = (end - start) / diff + 1;
        int[] answer = new int[len];
        for (int idx = 0, i = start; i <= end; idx++, i += diff)
            answer[idx] = num_list[i];
        return answer;
    }
}

File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.js
function solution(n, s, num_list) {
    return num_list.filter((it, i) => {
        switch (n) {
            case 1 :
                return (i <= s[1]);
            case 2 :
                return (s[0] <= i);
            case 3 :
                return (s[0] <= i && i <= s[1]);
            case 4 :
                return (s[0] <= i && i <= s[1] && (i - s[0]) % s[2] === 0);
        }
        return false;
    });
}